src/Features/ERDDAP/Platform/Observations/CurrentConditions/data_card.tsx

```diff
@@ -1,95 +1,158 @@
 import { UnitSystem } from "Features/Units/types"
 import { converter } from "Features/Units/Converter"
 
-import { PlatformFeatureWithDatasets, PlatformDataset } from "../../../types"
+import { useDataset } from "../../../hooks"
+import { PlatformFeature, PlatformTimeSeries } from "../../../types"
+
+import { cardProps, cardUrl } from "./common_card"
+import { NoDataCard } from "./no_data_card"
+import { LoadingDataCard } from "./loading_card"
+import { ErrorDataCard } from "./error_card"
 
 interface DataCardProps {
-  platform: PlatformFeatureWithDatasets
+  /** Data Type standard names to display */
   data_types: string[]
-  unit_system: UnitSystem
+  /** Platform */
+  platform: PlatformFeature
 }
 
-const cardProps = {
-  md: "4",
-  sm: "6",
-  style: {
-    paddingTop: "1rem",
-  },
-}
+/**
+ * Filters data types for a platform and fetches appropriate one
+ * to display based on preferential order and what is up to date
+ * @param data_types Data Type standard names to display
+ * @param platform Platform to display
+ */
+export const DataCard: React.FunctionComponent<DataCardProps> = ({ data_types, platform }) => {
+  const aDayAgo = new Date()
+  aDayAgo.setDate(aDayAgo.getDate() - 1)
 
-export const DataCard: React.SFC<DataCardProps> = ({ platform, data_types, unit_system }) => {
-  let filtered_datasets: PlatformDataset[] = []
+  let filteredTimeSeries: PlatformTimeSeries[] = []
 
-  data_types.forEach((data_type) => {
+  data_types.forEach((dataType) => {
     platform.properties.readings
-      .filter((reading) => data_type === reading.data_type.standard_name)
-      .forEach((reading) => filtered_datasets.push(reading))
+      .filter((reading) => dataType === reading.data_type.standard_name) // match any that are the current data type
+      .filter((reading) => (reading.time ? aDayAgo < new Date(reading.time) : false) && reading.depth < 2) // that have data in the last day, and that are near-surface
+      .forEach((reading) => filteredTimeSeries.push(reading))
   })
 
-  // Don't return anything if there are no datasets
-  if (filtered_datasets.length === 0) {
-    return null
+  // No current data or data types don't exist for platform
+  if (filteredTimeSeries.length < 1) {
+    let noFilteredTimeSeries: PlatformTimeSeries[] = []
+    data_types.forEach((dataType) => {
+      platform.properties.readings
+        .filter((reading) => dataType === reading.data_type.standard_name && reading.depth < 2)
+        .forEach((reading) => noFilteredTimeSeries.push(reading))
+    })
+
+    // Platform does not contain any of the requested data types
+    if (noFilteredTimeSeries.length < 1) {
+      return null
+    }
+
+    return <NoDataCard platform={platform} timeSeries={noFilteredTimeSeries[0]} />
   }
 
-  let reading = filtered_datasets[0]
+  const timeSeries = filteredTimeSeries[0]
 
-  let data = reading.readings
+  return <LoadDataCard timeSeries={timeSeries} platform={platform} />
```

Comment:
Codacy found an issue: ['LoadDataCard' was used before it was defined.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)